source/core/owt_base/RawTransport.cpp

```diff
@@ -225,7 +225,7 @@ void RawTransport<prot>::listenTo(uint32_t minPort, uint32_t maxPort)
                 m_socket.tcp.acceptor->open(tcp::v4());
                 m_socket.tcp.acceptor->bind(tcp::endpoint(tcp::v4(), port), ec);
 
-                if (ec != boost::system::errc::address_in_use) {
```

Comment:
If "ec" is "address_in_use" here, the loop will continue until successfully finding a port to bind.
Other "ec" conditions will be handled in line 228.
In which situation you want it to break in (!ec)?

Reply:
I think ec will be zero for success, all other error cases will return non zero value. Isn't the loop supposed to continue on error and come out of the loop as soon as there is a success or until the ports are exhausted?

Reply:
The loop is supposed to continue on address_in_use error, will break on success or other error.

Reply:
OK. I got the intention.

I tried to include some logs in the code, like below.  I am using 1.53.0 for boost. 

```
ELOG_WARN("TCP transport - checking port number : %d", port);
ELOG_WARN("TCP transport returned ec: %s, %d", ec.message().c_str(), ec.value());
if (ec != boost::system::errc::address_in_use) {
    ELOG_WARN("Break");
    break;
} 
```
Below are some of the logs from the modified code.

> 2020-05-27 08:22:35,480  - WARN: owt.RawTransport - TCP transport - checking port number : 30003
2020-05-27 08:22:35,480  - WARN: owt.RawTransport - TCP transport returned ec: Address already in use, 98
2020-05-27 08:22:35,480  - WARN: owt.RawTransport - Break
2020-05-27 08:22:35,480  - WARN: owt.RawTransport - TCP transport listen in port range bind error: Address already in use
2020-05-27 08:22:35,785  - WARN: owt.RawTransport - TCP transport - checking port number : 30000
2020-05-27 08:22:35,785  - WARN: owt.RawTransport - TCP transport returned ec: Success, 0
2020-05-27 08:22:35,785  - WARN: owt.RawTransport - Break

`if (ec != boost::system::errc::address_in_use)` condition doesn't seem to work always. 

Is there any dependency on boost version ?

Reply:
The error code log was "TCP transport returned ec: Address already in use, 98", but the "if (ec != boost::system::errc::address_in_use)" condition succeed? Did this happen always?

Reply:
I think if condition should be replaced with this instead.
 `if(ec.value() != boost::system::errc::address_in_use)`

Reply:
Hi, thanks for your suggestion.
Can you update your code in pull request so we can merge it?

Reply:
@starwarfan updated the request.